lib/provisioning/ProvisionDispatcher.js

```diff
@@ -0,0 +1,355 @@
+'use strict'; // eslint-disable-line
+
+const async = require('async');
+const crypto = require('crypto');
+const path = require('path');
+const zookeeper = require('node-zookeeper-client');
+
+const Logger = require('werelogs').Logger;
+
+function _genRandomHexChars(len) {
+    return crypto.randomBytes(Math.ceil(len / 2))
+        .toString('hex') // convert to hexadecimal format
+        .slice(0, len);  // return required number of characters
+}
+
+class ProvisionDispatcher {
+
+    /**
+     * Create a new provision dispatcher object
+     *
+     * @constructor
+     * @param {object} zkConfig - zookeeper config object
+
+     * @param {string} zkConfig.endpoint - zookeeper endpoint string
+     * where provisions are managed
+     * (e.g. "localhost:2181/shared-tasks")
+     * @param {object} logConfig - logging configuration
+     * @param {string} logConfig.logLevel - logging level
+     * @param {string} logConfig.dumpLevel - dump level
+     */
+    constructor(zkConfig, logConfig) {
+        this.zkEndpoint = zkConfig.endpoint;
+        this.client = zookeeper.createClient(zkConfig.endpoint);
+        this.client.connect();
+        this.connectWaitList = [];
+        this.client.once('connected', () => {
+            this.log.debug('connected to the ZK server');
+            this.connectWaitList.forEach(cb => cb());
+            this.connectWaitList = null;
+        });
+        this.myName = null;
+        this.myLeaderName = null;
+        this.isLeader = false;
+        this.owners = null;
+        this.provisions = null;
+
+        this.logger = new Logger('Backbeat:ProvisionDispatcher',
+                                 { level: logConfig.logLevel,
+                                   dump: logConfig.dumpLevel });
+        this.log = this.logger.newRequestLogger();
```

Comment:
Oh, ok, good to know, it indeed should be fixed.

How it should be done then? I'm not sure using the `logger` object works in this case, is there another call to use instead or is the `logger` object supposed to be used for logging calls?